packages/block-editor/src/store/selectors.js

```diff
@@ -1752,14 +1752,16 @@ export function canMoveBlock( state, clientId, rootClientId = null ) {
 	if ( attributes === null ) {
 		return true;
 	}
+	if ( getBlockEditingMode( state, rootClientId ) !== 'default' ) {
```

Comment:
Move the block editing check earlier so that blocks in the page editor won't be moved (which is expected.)